core/src/main/scala/kafka/server/DelayedElectLeader.scala

```diff
@@ -34,8 +34,8 @@ class DelayedElectLeader(
   responseCallback: Map[TopicPartition, ApiError] => Unit
 ) extends DelayedOperation(delayMs) {
 
-  var waitingPartitions = expectedLeaders
-  val fullResults = results.to[mutable.Set]
+  private var waitingPartitions = expectedLeaders
+  private val fullResults = mutable.Map() ++= results
```

Comment:
@hachikuji @jsancio Do you know why this was a `Set`? It seems to me that a `Map` makes more sense, but I thought I'd ask for a second opinion and I know you have made changes to related code recently.

Reply:
Agree, a `Map` makes more sense. Thanks for the change.